src/argilla/server/search_engine.py

```diff
@@ -185,15 +185,12 @@ async def search(
 
         text_query = self._text_query_builder(dataset, text=query.text)
 
-        bool_query: dict = {"must": [text_query]}
+        bool_query = {"must": [text_query]}
         if user_response_status_filter:
-            bool_query["filter"] = self._response_status_filter_builder(user_response_status_filter)
+            bool_query["should"] = self._response_status_filter_builder(user_response_status_filter)
```

Comment:
Since this is a filter, we don't want matches to be affected by the search score. Using the `should` clause will do this (see [docs](https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-bool-query.html#query-dsl-bool-query)). 

We can use the `filter` clause in combination with a `terms` filter. I will work on that.